### PR TITLE
Added ability to search multiple filenames

### LINKED ISSUE #3036: Multiple search criteria in `collect images` filename field
Currently the `filename` field of `collect images` only searches for one image that matches the criteria at once.

It would be great if we can enter in multiple search criteria - such as: 
8403 8405 8417

And it would show the three matching images.
It would be fine if it was separated by either a comma, or other symbol:
8403,8405,8417

### PR DESCRIPTION
**Pull request by sams96**
Added the ability to search for multiple keywords in the filenames collection using a comma seperated list. Fixes #3036. 

This could potentially be added to other collections but I thought I'd see if anyone would want that first.

### CODE REVIEW

**Comment by sams96** [comment]
I know you guys are busy getting ready for 3.0 and this is way too late to make it in but I thought I'd post it here now for some discussion until you're ready to start looking at features again.

**Comment by TurboGit** [comment]
Nice, simple and clean to me. Will be landing in 3.0.1 for sure. Thanks.

**Comment by parafin** [comment]
Can't this already be done using "add more images" in collect module?

**Comment by TurboGit** [comment]
@parafin : 6 max as there is only 6 rules.

**Comment by sams96** [comment]
@parafin this is also quite a bit quicker if you just want to search a bunch of filenames.

**Comment by sams96** [comment]
So I had a look at moving the loop outside the case/switch so it would work with all properties but it didn't work as easily as I thought. I did add a new tooltip to reflect the change I previously made though.

**Comment by sams96** [comment]
So I updated it so the matches for each name will stay in the collection module as you type the query, although I could make it so only the matches for the last name are shown. Also if you click one of the filenames it completely overwrites the query, I'm not totally sure what to do with that.

**Comment by TurboGit** [comment]
With just the matches of the last partial name displayed it does not look ok to me! This is really awkward.

**Comment by sams96** [comment]
@TurboGit it should be showing the matches for all the names, does this not work for you?

**Comment by TurboGit** [comment]
> @TurboGit it should be showing the matches for all the names, does this not work for you?

Only the names matching the last pattern.

**Comment by sams96** [comment]
It shows the matches for all the names for me, so I'm not really sure what I can do about it.

**Comment by TurboGit** [comment]
Ok, I just tested again and this is indeed working now. I may have done a procedural error will testing. Thanks!.

Please remember that we don't want to have master merged into topic branches in PR. Better to rebase than merging master.

**Comment by TurboGit** [comment]
Merged on master and 3.0.1.

File: src/common/collection.c [hunk 1]
```diff
@@ -1446,8 +1446,17 @@ static gchar *get_query_string(const dt_collection_properties_t property, const
...
-      query = dt_util_dstrcat(query, "(filename LIKE '%%%s%%')", escaped_text);
+    {
+      GList *list, *l;
+      list = dt_util_str_to_glist(",", escaped_text);
+
+      for (l = list; l != NULL; l = l->next)
+        l->data = dt_util_dstrcat(query, "(filename LIKE '%%%s%%')", (char *)l->data);
+
+      query = dt_util_glist_to_str(" OR ", list);
+
```
Comment (TurboGit) [inline src/common/collection.c]: we need to free list here : g_list_free(list);

File: src/libs/collect.c [hunk 2]
```diff
@@ -637,6 +637,23 @@ static gboolean list_match_string(GtkTreeModel *model, GtkTreePath *path, GtkTre
...
+      if(g_str_has_prefix((char *)l->data, "%"))
+      {
+        if((visible = (g_strrstr(haystack, (char *)l->data + 1) != NULL))) break;
+      }
+      else
+      {
+        if((visible = (g_strrstr(haystack, (char *)l->data) != NULL))) break;
+      }
+    }
+  }
```
Comment (TurboGit) [inline src/libs/collect.c]: free list here too.

**Pull request merged**